kernel/Dockerfile

```diff
@@ -98,8 +101,14 @@ RUN make defconfig && \
     cp System.map /out && \
     ([ -n "${DEBUG}" ] && cp vmlinux /out || true)
 
+# WireGuard
+RUN git clone --depth 1 -b "$WIREGUARD_TAG" "$WIREGUARD_REPO" /wireguard && \
```

Comment:
You'll notice here that I'm using doing a shallow clone, since all we really want is the tag. Then in the next line, I verify that the tag has a valid GPG signature. This method works fine.

However, since we're using `WIREGUARD_TAG` now, instead of my fancy auto-latest magic of the last commit, it means we have to update this file for snapshot bumps. That's fine. But it does raise the question: why are we even using git at all? The WireGuard upstream project releases tarballs of these tags along with their checksums in a GPG signed email. It'd probably be more efficient to just download this tarball and check the checksum. Updates would then update not only the `WIREGUARD_TAG` but also the `WIREGUARD_HASH` variable.

Would you prefer this? It makes more sense to me anyhow.